adaptor/adaptor.go

```diff
@@ -32,6 +45,13 @@ type Adaptor interface {
 	Stop() error
 }
 
```

Comment:
I _think_ this will pay more dividends down the road when we start passing a `context.Context` around.